testing.md

```diff
@@ -0,0 +1,299 @@
+---
+layout: page
+title: Testing
+permalink: /testing/
+---
+
+* TOC
+{:toc}
+
+## Introduction
+
+The more features your application has, the harder it is to test it manually. A
+good set of automated tests will help you make sure your application performs
+correctly when you publish it to users while retaining your feature and bug fix
+velocity.
+
+There are many kinds of automated testing. Their definitions are fuzzy, so let's
+define them here for clarity:
+
+- A _unit test_ tests a single function, method or class. External dependencies
+  of the test subject are generally mocked out using, for example,
+  [`package:mockito`](https://github.com/fibulwinter/dart-mockito/blob/master/README.md).
+  Unit tests generally do not read from/write to disk, render to screen and do
+  not receive user actions from outside the process running the test. The goal
+  of a unit-test is to verify the correctness of a piece of logic under a
+  variety of conditions.
+- A _widget test_ (in other UI frameworks referred to as _component test_) tests
+  a single widget. Testing a widget involves multiple classes and requires a
```

Comment:
Widget tests still generally don't read or write files, render to screen, or involve user actions from outside the test. From that perspective they're very similar to unit tests.